ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm

```diff
@@ -0,0 +1,298 @@
+#include "../PrjFSKext/kernel-header-wrappers/vnode.h"
+#include "../PrjFSKext/KauthHandlerTestable.hpp"
+#include "../PrjFSKext/PerformanceTracing.hpp"
+#include "../PrjFSKext/VirtualizationRootsTestable.hpp"
+#include "MockProc.hpp"
+#include "MockVnodeAndMount.hpp"
+#include "VnodeCacheEntriesWrapper.hpp"
+
+#import <XCTest/XCTest.h>
+
+using std::shared_ptr;
+
+// Dummy implementation
+class org_vfsforgit_PrjFSProviderUserClient
+{
+};
+
+@interface ShouldHandleFileOpTests : XCTestCase
+@end
+
+@implementation ShouldHandleFileOpTests
+{
+    VnodeCacheEntriesWrapper cacheWrapper;
+    vfs_context_t context;
+    PerfTracer perfTracer;
+    shared_ptr<mount> testMount;
+    std::string repoPath;
+    shared_ptr<vnode> repoRootVnode;
+    shared_ptr<vnode> testVnodeFile;
+}
+
+- (void) setUp {
+    kern_return_t initResult = VirtualizationRoots_Init();
+    XCTAssertEqual(initResult, KERN_SUCCESS);
+    self->cacheWrapper.AllocateCache();
+    self->context = vfs_context_create(nullptr);
+    MockProcess_AddContext(self->context, 501 /*pid*/);
+    MockProcess_SetSelfPid(501);
+    MockProcess_AddProcess(501 /*pid*/, 1 /*credentialId*/, 1 /*ppid*/, "test" /*name*/);
+    
+    self->testMount = mount::Create();
+    self->repoPath = "/Users/test/code/Repo";
+    self->repoRootVnode = self->testMount->CreateVnodeTree(self->repoPath, VDIR);
+    self->testVnodeFile = self->testMount->CreateVnodeTree(self->repoPath + "/file.txt");
+}
+
+- (void) tearDown {
+    self->testVnodeFile.reset();
+    self->repoRootVnode.reset();
+    self->testMount.reset();
+    MockProcess_Reset();
+    self->cacheWrapper.FreeCache();
+    MockVnodes_CheckAndClear();
+    VirtualizationRoots_Cleanup();
+}
+
+- (void)testRootHandle {
+    int pid;
+    VirtualizationRootHandle testRootHandle;
+    
+    // Invalid Root Handle Test
+    XCTAssertFalse(
+        ShouldHandleFileOpEvent(
+            &self->perfTracer,
+            self->context,
+            self->repoRootVnode.get(),
+            nullptr, // path
+            KAUTH_FILEOP_RENAME,
+            true, // isDirectory,
+            &testRootHandle,
+            &pid));
+
+    PrjFSProviderUserClient userClient;
```

Comment:
There's probably no harm moving this to an ivar - various other tests also use one of these.